src/main/java/com/algorand/sdkutils/listeners/JavaGenerator.java

```diff
@@ -378,20 +379,28 @@ public void storeEnumDefinition(TypeDef typeObj) {
             String javaEnum = Tools.getCamelCase(val, true).toUpperCase();
             sb.append(javaEnum);
             sb.append("(\"" + val + "\")");
-            if (elmts.hasNext()) {
-                sb.append(",\n");
-            } else {
-                sb.append(";\n\n");
-            }
+            sb.append(",\n");
         }
+        sb.append(TAB + TAB + "@JsonProperty(\"\") UNKNOWN(\"\");\n\n");
+
         sb.append(TAB + TAB + "final String serializedName;\n");
         sb.append(TAB + TAB + javaTypeName + "(String name) {\n");
         sb.append(TAB + TAB + TAB + "this.serializedName = name;\n");
         sb.append(TAB + TAB + "}\n\n");
         sb.append(TAB + TAB + "@Override\n");
         sb.append(TAB + TAB + "public String toString() {\n");
         sb.append(TAB + TAB + TAB + "return this.serializedName;\n");
-        sb.append(TAB + TAB + "}\n");
+        sb.append(TAB + TAB + "}\n\n");
+
+        sb.append(TAB + TAB + "@JsonCreator\n");
+        sb.append(TAB + TAB + "public static " + javaTypeName + " forValue(String value) {\n");
+        sb.append(TAB + TAB + TAB + "for (" + javaTypeName + " t : values()) {\n");
+        sb.append(TAB + TAB + TAB + TAB + "if (t.serializedName.equalsIgnoreCase(value)) {\n");
+        sb.append(TAB + TAB + TAB + TAB + TAB + "return t;\n");
+        sb.append(TAB + TAB + TAB + TAB + "}\n");
+        sb.append(TAB + TAB + TAB + "}\n");
+        sb.append(TAB + TAB + TAB + "return UNKNOWN;\n");
```

Comment:
> * To minimize chance of a partial future update, can a constant be created for UNKNOWN?
`UNKNOWN` is an enum type defined in each type, hopefully this is clearer in the SDK PR.

> * To avoid regressions, can a test be added to generator proving default model generation defends against unsupported enum values?
I'm not sure what that test would look like in generator, but I have such a test in the SDK.

Reply:
@winder Thanks for the reply - I'm OK to resolve.  Leaving a note for completeness.

* I'm trying to say the literal `UNKNOWN` is referenced in `JavaGenerator.java` multiple times.  If the generator is later changed, it's possible someone partially changes references.  A partial change _should_ be caught by downstream generation (or a test in _generator_), so I don't feel strongly.
* In _generator_, I think it's possible to define a test model (or use `petstore.json`) to generate Java (among other) types and assert expected behavior.
  * I see you're performing an analog to what I'm describing via `testHttpModelEnum` in java-algorand-sdk.  I think a case can be made that `testHttpModelEnum` _and_ an analogous generator test should _both_ exist.  
  * I'd typically go one step further and perform round-trip serialization testing.  The test tests deserialization, but _not_ serialization.  I'll leave a note in java-algorand-sdk.